content/posts/git/webhooks/index.md

```diff
@@ -0,0 +1,59 @@
+---
+date: "2015-06-09"
+title: Les webhooks
+tags:
+  - git
+  - webhooks
+authors:
+  - magsout
+header:
+  credit: https://www.flickr.com/photos/haru__q/14396323454/
+
+---
+
+
+Si vous utilisez GitHub pour héberger vos sources et que vous cherchez un moyen très simple (peut-être un peu trop) de mettre en place le déploiement continue _sans serveur d'intégration continue dédié_, alors les webhooks peuvent répondre parfaitement à votre besoin.
+
+## Les quoi ?
+
+Les webhooks permettent de lancer une action de façon automatique par l'intermédiaire d'un ou plusieurs événements depuis un dépôt [GitHub](GitHub.com). Concrètement il va être possible de déployer son projet sur un serveur de production, sauvegarder son site internet, ou même mettre à jour des stats des qu'un commit sera envoyé ou même une issue ouverte. Il est possible de faire tout ce qu'on veut dans la limite de son imagination comme l'explique Github.
+
+## Comment ça marche ?
+
+Le fonctionnement est on ne peut plus simple, un événement (merge, push, release etc..) va déclencher une requête HTTP POST vers l'url que vous avez configurée. Le reste de l'exécution, se fera sur le serveur où pointe l'url, libre à vous d'écrire le script que vous voulez pour effectuer les opérations que vous souhaitez.
+
+Il faut cependant être contient que la simplification a un coût, à savoir que même si une PR, un commit venaient à rendre inopérant votre projet l'événement se lancerait quand même.
+
+## La mise en place
+
+Seul le propriétaire d'un dépôt dispose des droits nécessaires pour configurer un webhook. Il suffit pour cela de se rendre dans `Settings` puis `Webhooks & Services` :
+
+![](setting_webhook.jpg)
+
+Un petit clic sur `Add webhook` et vous obtenez une interface encore une fois très simple :
+
+![](configuration_webhook.jpg)
+
+Cette interface se compose de trois parties :
+
+- 1) l'URL vers laquelle votre requête sera lancée
```

Comment:
You can use markdow numbered list with

```md
1. one
- two
- three
```

Result (see raw source of this comment please)

1. One 
- two
- three